test/bio_pw_callback_test.c

```diff
@@ -0,0 +1,405 @@
+/*
+ * Copyright 2024 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include "testutil.h"
+
+#include <openssl/bio.h>
+#include <openssl/pem.h>
+
+/* dummy data that needs to be passed to the callback */
+typedef struct CallbackData {
+    char magic;
+    int result;
+} CALLBACK_DATA;
+
+/* constants */
+static char *key_password = "weak_password";
+static int key_password_len = 13;
+static char *a0a_password = "aaaaaaaa\0aaaaaaaa";
+static int a0a_password_len = 17;
+static char *a0b_password = "aaaaaaaa\0bbbbbbbb";
+static int a0b_password_len = 17;
+static char cb_magic = 'p';
```

Comment:
```suggestion
static const char cb_magic = 'p';
```

Please use `const` for all of these constants.